src/Native/System.Native/pal_console.cpp

```diff
@@ -35,9 +35,10 @@ extern "C" int32_t GetWindowSize(WinSize* windowSize)
 #endif
 }
 
-extern "C" int32_t IsATty(int fd)
+extern "C" int32_t IsATty(intptr_t fd)
 {
-    return isatty(fd);
+    assert(INT32_MIN <= fd && fd <= INT32_MAX);
```

Comment:
IMHO these shouldn't be asserts in any case. If we're extending the potential range of inputs, we should be validating that input, setting `errno`, and returning something that indicates failure when necessary.